compiler/rustc_middle/src/ty/context.rs

```diff
@@ -66,8 +66,8 @@ use std::mem;
 use std::ops::{Bound, Deref};
 use std::sync::Arc;
 
-/// A type that is not publicly constructable. This prevents people from making `TyKind::Error`
-/// except through `tcx.err*()`, which are in this module.
+/// A type that is not publicly constructable. This prevents people from making [`TyKind::Error`]s
+/// except through the error-reporting functions on a [`tcx`][TyCtxt].
 #[derive(Copy, Clone, Debug, Eq, Hash, PartialEq, PartialOrd, Ord)]
 #[derive(TyEncodable, TyDecodable, HashStable)]
 pub struct DelaySpanBugEmitted(());
```

Comment:
Also, why does `DelaySpanBugEmitted` contain the unit type? Isn't this semantically equivalent to:

```suggestion
pub struct DelaySpanBugEmitted;
```

Reply:
No, the latter would be constructible outside this module, which the type is trying to prevent

Reply:
Ah, that makes sense. Thanks!